pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -178,6 +178,93 @@ func (ic *ifConfigurator) configureContainerLinkSriov(
 	return nil
 }
 
+// configureContainerLinkSriovInterface moves VF to the container namesapce;
+// Intended for multiple interfaces other than primary interface
+func (ic *ifConfigurator) configureContainerLinkSriovInterface(
+        podName string,
+        podNamespace string,
+        containerID string,
+        containerNetNS string,
+        containerIfaceName string,
+        mtu int,
+        pciAddress string,
+        result *current.Result,
+) error {
+        hostIface := &current.Interface{Name: containerIfaceName}
+        containerIface := &current.Interface{Name: containerIfaceName, Sandbox: containerNetNS}
+        result.Interfaces = []*current.Interface{hostIface, containerIface}
+
+        if pciAddress != "" {
+                // Get rest of the VF information
+                pfName, vfID, err := util.GetVfInfo(pciAddress)
+		klog.V(2).Infof("pfName and vfID of pciAddress: %+v, %+v, %s", pfName, vfID, pciAddress)
+                if err != nil {
+			return fmt.Errorf("configureContainerLinkSriovInterface: failed to get VF information: %+v", err)
+                }
+        } else {
+		return fmt.Errorf("configureContainerLinkSriovInterface: VF pci address is required")
+        }
+
+        vfIFName, err := util.GetVFLinkNames(pciAddress)
+        if err != nil || vfIFName == "" {
+		return fmt.Errorf("configureContainerLinkSriovInterface: VF interface not found for pciAddress %s: %+v", pciAddress, err)
+        }
+
+        link, err := netlink.LinkByName(vfIFName)
+        klog.V(2).Infof("get link of vfIFName: %+v, link: %+v", vfIFName, link)
+        if err != nil {
+		return fmt.Errorf("configureContainerLinkSriovInterface: error getting VF link: %+v", err)
+        }
+
+        hostIface.Mac = link.Attrs().HardwareAddr.String()
+        hostIface.Name = vfIFName
+        klog.V(2).Infof("hostIface.Name: %+v, hostIface.Mac: %+v, vfIFName: %+v", hostIface.Name, hostIface.Mac, vfIFName)
+
+        // Move VF to Container namespace
+        netns, err := ns.GetNS(containerNetNS)
+        if err != nil {
+               return fmt.Errorf("configureContainerLinkSriovInterface: failed to open container netns %s: %+v", containerNetNS, err)
+        }
+        err = moveIfToNetns(vfIFName, netns)
```

Comment:
Do you think we can extract shared code (line 228 - 260) with configureContainerLinkSriov to a separate func? Or maybe we can even share code with configureContainerLinkVeth(), but I did not look into the code to be sure.

Reply:
Done